src/os/ui/im/action/filteraction.js

```diff
@@ -165,24 +165,25 @@ const onEditComplete = function(original, entry) {
   // don't do anything if there was no change
   if (entry && (!original || entry.compare(original) !== 0)) {
     var cmds = [];
-
     var iam = getImportActionManager();
-    var entries = iam.getActionEntries(entry.getType());
-
     var entryTitle = entry.getTitle();
     var insertIndex;
     var parentId;
+    var searchArray;
 
     if (original) {
-      insertIndex = olArray.findIndex(entries, function(entry) {
-        return entry == original;
-      });
-
       var parent = original.getParent();
       if (parent) {
         parentId = parent.getId();
+        searchArray = parent.getChildren();
+      } else {
+        searchArray = iam.getActionEntries(entry.getType());
       }
 
+      insertIndex = olArray.findIndex(searchArray, function(entry) {
```

Comment:
Nit: `findIndex` is now a natively supported function on arrays, so the preferred way to do this would be `searchArray.findIndex(...)` rather than using the OL utility function. We prefer to clean up old uses of functions like this in favor of native functions whenever possible these days.